src/data/climate.ts

```diff
@@ -15,12 +14,34 @@ export type HvacMode =
 
 export const CLIMATE_PRESET_NONE = "none";
 
-type ClimateAttributes = TranslationDict["state_attributes"]["climate"];
-export type HvacAction = keyof ClimateAttributes["hvac_action"];
-export type FanMode = keyof ClimateAttributes["fan_mode"];
+export type HvacAction =
+  | "off"
+  | "heating"
+  | "cooling"
+  | "drying"
+  | "idle"
+  | "fan";
+export type FanMode =
+  | "off"
+  | "on"
+  | "auto"
+  | "low"
+  | "medium"
+  | "high"
+  | "top"
+  | "middle"
+  | "focus"
+  | "diffuse";
 export type PresetMode =
-  | keyof ClimateAttributes["preset_mode"]
-  | typeof CLIMATE_PRESET_NONE;
+  | "none"
+  | "eco"
+  | "away"
+  | "boost"
+  | "comfort"
+  | "home"
+  | "sleep"
+  | "activity";
```

Comment:
@emontnemery May be we can just use `string` for attributes instead of enum because integrations can override preset, fan and swing mode with anything.

Reply:
That is correct, presets are not pre-defined and are free strings.

Reply:
But not swing modes and hvac actions?

Reply:
Correct, hvac modes, hvac actions, and swing modes are limited by design

- ref: <https://developers.home-assistant.io/docs/core/entity/climate#hvac-modes>
- ref: <https://developers.home-assistant.io/docs/core/entity/climate#swing-modes>